core/templates/services/math-interactions.service.ts

```diff
@@ -133,7 +149,20 @@ export class MathInteractionsService {
     return true;
   }
 
-  validateEquation(equationString: string): boolean {
+  validateAlgebraicExpression(
+      expressionString: string, validVariablesList: Array<string>) {
+    if (validVariablesList === undefined) {
```

Comment:
No, please don't do this, @aks681 already pointed it out. Have every caller explicitly pass things in.

@nishantwrp -- a question for you. Shouldn't this sort of thing be caught by the typescript checks? /cc @vojtechjelinek

Reply:
Nope! typescript doesn't throw an error on comparing an argument with `undefined` https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABADwBQEMBciDOUBOMYA5gJSIDeAUIrYjMIhogLxuLgAmApsEd53LU6IxOhp0AvlUlA

Reply:
Hi @nishantwrp -- no, what I mean is, shouldn't it prevent an undefined validVariablesList from being passed in, in the first place?

Reply:
Are you saying that can we call this function as

 validateAlgebraicExpression('some string', undefined);


then yes. we can assign null or undefined to any type.

Reply:
Oh, that's annoying. Is there any setting to forbid that, in either the linter or TS checks? I can understand null, but we should prohibit undefined. /cc @vojtechjelinek

Reply:
(E.g. if a key in an object is missing, it's basically undefined, so if the TS checks are not checking for this then their utility is somewhat decreased.)

Reply:
Relevant? https://mariusschulz.com/blog/non-nullable-types-in-typescript#strict-null-checking

Either we do it for both null + undefined, or we do it for just undefined, but this is one of the sorts of things that we ought to be catching.

Reply:
Yup, I agree we can enable this check.

Reply:
> (E.g. if a key in an object is missing, it's basically undefined, so if the TS checks are not checking for this then their utility is somewhat decreased.)

I think ts would catch this.

Reply:
Thanks @nishantwrp. Filed https://github.com/oppia/oppia/issues/10327 and assigned to you.

Reply:
(And @iamprayush -- just noting that you still have an action item on this comment thread: https://github.com/oppia/oppia/pull/10269#discussion_r470224163, despite the conversation after that not being relevant to this PR. Thanks!)

Reply:
Yup, I've removed the undefined check completely now.